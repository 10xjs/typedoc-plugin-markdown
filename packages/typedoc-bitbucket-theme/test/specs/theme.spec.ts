import { TestApp } from 'typedoc-plugin-markdown/test/test-app';

let testApp: TestApp;

beforeAll(() => {
  testApp = new TestApp(['theme.ts']);
});
describe(`Theme:`, () => {
  test(`should getUrls by path'`, () => {
    testApp.bootstrap({ theme: './dist' });
    const urlMappings = testApp.theme.getUrls(testApp.project);
    expect(TestApp.getExpectedUrls(urlMappings)).toMatchSnapshot();
  });
  test(`should getUrls by name'`, () => {
    testApp.bootstrap({ theme: 'bitbucket' });
    const urlMappings = testApp.theme.getUrls(testApp.project);
    expect(TestApp.getExpectedUrls(urlMappings)).toMatchSnapshot();
  });
});
