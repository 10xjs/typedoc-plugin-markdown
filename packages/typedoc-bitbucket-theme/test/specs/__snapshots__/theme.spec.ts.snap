// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme: should getUrls by name' 1`] = `
Array [
  "globals.md",
  "README.md",
  "modules/_theme_.md",
  "modules/_theme_.md#markdown-header-declarationitema",
  "modules/_theme_.md#markdown-header-declarationitemb",
  "modules/_theme_.md#markdown-header-functionitema",
  "modules/_theme_.md#markdown-header-functionitemb",
  "modules/_theme_.namespacea.md",
  "modules/_theme_.namespaceb.md",
  "enums/_theme_.enumitema.md",
  "enums/_theme_.enumitemb.md",
  "classes/_theme_.classitema.md",
  "classes/_theme_.classitemb.md",
  "interfaces/_theme_.interfaceitema.md",
  "interfaces/_theme_.interfaceitema.md#markdown-header-prop",
  "interfaces/_theme_.interfaceitemb.md",
  "interfaces/_theme_.interfaceitemb.md#markdown-header-prop",
]
`;

exports[`Theme: should getUrls by path' 1`] = `
Array [
  "globals.md",
  "README.md",
  "modules/_theme_.md",
  "modules/_theme_.md#markdown-header-declarationitema",
  "modules/_theme_.md#markdown-header-declarationitemb",
  "modules/_theme_.md#markdown-header-functionitema",
  "modules/_theme_.md#markdown-header-functionitemb",
  "modules/_theme_.namespacea.md",
  "modules/_theme_.namespaceb.md",
  "enums/_theme_.enumitema.md",
  "enums/_theme_.enumitemb.md",
  "classes/_theme_.classitema.md",
  "classes/_theme_.classitemb.md",
  "interfaces/_theme_.interfaceitema.md",
  "interfaces/_theme_.interfaceitema.md#markdown-header-prop",
  "interfaces/_theme_.interfaceitemb.md",
  "interfaces/_theme_.interfaceitemb.md#markdown-header-prop",
]
`;
